mathematica/ieee754_floating_point_evaluation.wl

```diff
@@ -0,0 +1,64 @@
+(* ::Package:: *)
+
+BeginPackage["IEEE754FloatingPointEvaluation`"]
+
+
+Get[FileNameJoin[{NotebookDirectory[],"ieee754_floating_point.wl"}]]
+
+
+IEEEEvaluate;
+IEEEEvaluate::usage =
+"IEEEEvaluate[\!\(\*StyleBox[\"x\",\nFontSlant->\"Italic\"]\)] " <>
+"evaluates \!\(\*StyleBox[\"x\",\nFontSlant->\"Italic\"]\) " <>
+"according to the rules of IEEE arithmetic.  It uses the format set by " <>
+"SetFloatingPointFormat and the rounding mode set by SetRoundingMode.";
+IEEEEvaluate::argnum =
+"IEEEEvaluate called with `1` arguments; 1 argument is expected.";
+IEEEEvaluate::badarg =
+"IEEEEvaluate must be called with Plus, Times, Power, or a number."
+
+
+UseFMA;
+UseFMA::usage =
+"UseFMA is an option for IEEEEvaluate that specifies whether to use FMA " <>
+"for expressions of the form \!\(\*
+StyleBox[\"a\",\nFontSlant->\"Italic\"]\) * \!\(\*
+StyleBox[\"b\",\nFontSlant->\"Italic\"]\) + \!\(\*
+StyleBox[\"c\",\nFontSlant->\"Italic\"]\).";
+
+
+Begin["`Private`"]
+
+
+SetAttributes[IEEEEvaluate,HoldAll];
+Options[IEEEEvaluate]={UseFMA->True};
+IEEEEvaluate[x:(_Plus|_Times|_Power|_?NumberQ),OptionsPattern[]]:=
+Block[
+{Plus,Times,ev},
+ClearAttributes[Plus,Flat];
+ClearAttributes[Times,Flat];
+SetAttributes[ev,HoldAll];
+ev[a_*b_+c_]:=
+If[
+OptionValue[UseFMA],
+CorrectlyRound[IEEEEvaluate[a]IEEEEvaluate[b]+IEEEEvaluate[c]],
+CorrectlyRound[CorrectlyRound[IEEEEvaluate[a]IEEEEvaluate[b]]+IEEEEvaluate[c]]];
+ev[a_ +b_]:=CorrectlyRound[IEEEEvaluate[a]+IEEEEvaluate[b]];
```

Comment:
```suggestion
ev[a_ +b_]:=CorrectlyRound[IEEEEvaluate[a]+IEEEEvaluate[b]];
ev[a_ +b__]:=(Message[IEEEEvaluate::badarg]; $Failed);
```

```mathematica
SetFloatingPointFormat[binary64]
SetRoundingMode[NearestTiesToEven]
Nest[CorrectlyRound[#+CorrectlyRound[1/10]]&,0,10]
```
> 9007199254740991/9007199254740992

But
```mathematica
IEEEEvaluate[1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10,UseFMA->False]
```
> 1

This works:
```mathematica
IEEEEvaluate[((((((((1/10+1/10)+1/10)+1/10)+1/10)+1/10)+1/10)+1/10)+1/10)+1/10,UseFMA->False]
```
> 9007199254740991/9007199254740992

But `IEEEEvaluate[1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10+1/10,UseFMA->False]` should be an error.

Add these tests to the _test.wl.